docs/decisions/broker-trigger.md

```diff
@@ -0,0 +1,137 @@
+# Broker and Trigger model (for 0.5.0)
+
+Decision date: 6 March 2019
+
+## Define High-Level Objects Related to Eventing
+
+The goal of eventing is to provide an abstract "bucket of events" which event
+consumers can sample from using filters on CloudEvent
+[context attributes](https://github.com/cloudevents/spec/blob/master/spec.md#context-attributes).
+Event producers can deliver events into the bucket without needing to understand
+the exact mechanics of routing to specific consumers. Event producers and
+consumers are decoupled temporally but also through a deliberately black-box
+abstraction which can optimize the underlying routing layer dynamically.
+
+## Objects
+
+The following objects are the minimum set needed to experiment and collect
+customer feedback on the Eventing object model, aka an MVP (including rough YAML
+definitions; final naming and field definitions subject to change based on
+implementation experience).
+
+### Broker
+
+A Broker represents a "bucket of events". It is a namespaced object, with
+automation provided to provision a Broker named `default` in
+appropriately-labelled namespaces. It is expected that most users will not need
+to explicitly create a Broker and simply enabling eventing in the namespace to
+create the `default` Broker will be sufficient. Assuming that the Broker model
+is successful, future recommendations for Source CRD authors would be to default
+the Source's `spec.sink` attribute to point to the `default` Broker in the
+namespace.
+
+Historical storage and replay of events is not part of the Broker design or MVP.
+
+Broker implements the "Addressable" duck type to allow Source custom resources
+to target the Broker. Broker DNS names should also be predictable (e.g.
+`default-broker` for the `default` Broker), to enable whitebox event producers
```

Comment:
Does this implies that the "default" broker MUST always be present? e.g. if someone deletes it the system will recreate it?

Reply:
There's a k8s watch set up on the namespaces and if the appropriate label is found there, then the Broker is created. So if a user manually deletes the Broker it will be recreated.

Reply:
I don't read this as requiring the `default` Broker to be present. Rather, that when a whitebox event producer is given a Broker to send events to, they can fully construct the DNS name for that Broker, without needing to get the Broker object itself.

From earlier in the doc:
> It is expected that most users will not need to explicitly create a Broker and simply enabling eventing in the namespace to create the `default` Broker will be sufficient.

So we _think_ that most of the time it will be the `default` Broker, but nothing requires it to be. Nor does anything require the `default` Broker to exist.

Reply:
Added a mention earlier about the mechanism for provisioning the `default` broker, which is (as Ville suggests) that the broker is created by reconciling the namespaces with the appropriate annotation.